pilot/pkg/serviceregistry/kube/controller/util.go

```diff
@@ -104,8 +104,10 @@ func findServiceTargetPort(servicePort *model.Port, k8sService *v1.Service) (int
 		if p.Name == servicePort.Name || p.Port == int32(servicePort.Port) {
 			if p.TargetPort.Type == intstr.Int && p.TargetPort.IntVal > 0 {
 				targetPort = int(p.TargetPort.IntVal)
-			} else {
+			} else if p.TargetPort.StrVal != "" {
 				targetPortName = p.TargetPort.StrVal
+			} else {
```

Comment:
I guess this could never happen, let me verify

Reply:
Yes, this branch can never enter, because k8s will by default set the targetPort = port if user does not set it.

Reply:
I see – thanks for clarifying. Is the behavior in the description correct though? I was really confused when I tried to do it myself.

Reply:
Yes, it is expected